v2/anomalydetection/backend-setup.mdx

```diff
@@ -0,0 +1,683 @@
+        maxRequests: [
+            {
+                limit: 100,
+                perTimeIntervalMS: 1000,
+            },
+        ]
+    }
+]
+```
+
+- `passwordHash`: The SHA-1 hash of the password that needs to be checked against the breach database. If this is not provided, the password breach check will be skipped.
+- `email`: The email address that is being used for the authentication event. If this is not provided (or `phoneNumber`), the impossible travel detection, new device detection and device count detection will be skipped.
+- `phoneNumber`: The phone number that is being used for the authentication event. If this is not provided (or `email`), the impossible travel detection, new device detection and device count detection will be skipped.
+- `actionType`: The type of action that is being performed. The possible values are:
+      - "emailpassword-sign-in"
+      - "emailpassword-sign-up"
+      - "send-password-reset-email"
+      - "passwordless-send-email"
+      - "passwordless-send-sms"
+      - "totp-verify-device"
+      - "totp-verify-totp"
+      - "thirdparty-login"
+      - "emailverification-send-email"
+
+
+Below is an example of how to make the request to the anomaly detection endpoint:
+
+```tsx
+import axios from "axios";
+import { createHash } from 'crypto';
+
+const API_KEY = "<api-key>"; // Your secret API key that you received from the SuperTokens team
+const ANOMALY_DETECTION_API_URL = "https://security-us-east-1.aws.supertokens.io/v1/security";
+
+async function doSecurityChecks(input: {
+    actionType?: string,
+    email?: string,
+    phoneNumber?: string,
+    password?: string,
+    requestId?: string,
+    bruteForce?: {
+        keyStr: string,
+        maxRequests: {
+            limit: number,
+            perTimeIntervalMS: number
+        }[]
+    }[]
+}): Promise<{
+    status: "GENERAL_ERROR",
+    message: string
+} | undefined> {
+    let requestBody: {
+        email?: string;
+        phoneNumber?: string;
+        actionType?: string;
+        requestId?: string;
+        passwordHash?: string;
+        bruteForce?: {
+            keyStr: string;
+            maxRequests: {
+                limit: number;
+                perTimeIntervalMS: number;
+            }[];
+        }[];
+    } = {}
+
+    if (input.requestId !== undefined) {
+        requestBody.requestId = input.requestId;
+    }
+
+    if (input.password !== undefined) {
+        let shasum = createHash('sha1');
+        shasum.update(input.password);
+        const passwordHash = shasum.digest('hex');
+        requestBody.passwordHash = passwordHash;
+    }
+    requestBody.bruteForce = input.bruteForce;
+    requestBody.email = input.email;
+    requestBody.phoneNumber = input.phoneNumber;
+    requestBody.actionType = input.actionType;
+
+    let response = await axios.post(ANOMALY_DETECTION_API_URL, requestBody);
+
+    let responseData = response.data;
+    if (!responseData.bruteForce.result) {
```

Comment:
also add check for bot detection, and if it's a vpn or tor IP

Reply:
done